python/dlr/compatibility.py

```diff
@@ -0,0 +1,208 @@
+        vindex = int(item["attrs"]["var"])
+        item["attrs"]["name_hint"] = nodes[vindex]["attrs"]["name"]
+        # set vindex to null
+        nodes[vindex]["type_key"] = ""
+        del item["attrs"]["var"]
+        assert item["type_key"].startswith("relay.")
+        item["type_key"] = item["type_key"][len("relay."):]
+        return item
+
+    def _rename(new_name):
+        def _convert(item, _):
+            item["type_key"] = new_name
+            return item
+        return _convert
+
+    def _update_tir_var(new_name):
+        def _convert(item, _):
+            item["type_key"] = new_name
+            item["attrs"]["type_annotation"] = "0"
+            return item
+        return _convert
+
+    def _update_global_key(item, _):
+        item["repr_str"] = item["global_key"]
+        del item["global_key"]
+        return item
+
+    def _update_op(item, _):
+        item["repr_str"] = item["global_key"]
+        del item["global_key"]
+        if item["repr_str"] == "contrib.adaptive_avg_pool2d":
+            item["repr_str"] = "nn.adaptive_avg_pool2d"
+        elif item["repr_str"] == "contrib.adaptive_max_pool2d":
+            item["repr_str"] = "nn.adaptive_max_pool2d"
+        return item
+
+    def _update_resize_attrs(item, _):
+        if "align_corners" in item["attrs"]:
+            if item["attrs"]["align_corners"] == "1":
+                item["attrs"]["coordinate_transformation_mode"] = "align_corners"
+            else:
+                item["attrs"]["coordinate_transformation_mode"] = "half_pixel"
+            del item["attrs"]["align_corners"]
+        return item
+
+    node_map = {
+        # Base IR
+        "SourceName": _update_global_key,
+        "EnvFunc": _update_global_key,
+        "relay.Op": _update_op,
+        "relay.TypeVar": _ftype_var,
+        "relay.GlobalTypeVar": _ftype_var,
+        "relay.Type": _rename("Type"),
+        "relay.TupleType": _rename("TupleType"),
+        "relay.TypeConstraint": _rename("TypeConstraint"),
+        "relay.FuncType": _rename("FuncType"),
+        "relay.IncompleteType": _rename("IncompleteType"),
+        "relay.TypeRelation": _rename("TypeRelation"),
+        "relay.TypeCall": _rename("TypeCall"),
+        "relay.Module": _rename("IRModule"),
+        "relay.SourceName": _rename("SourceName"),
+        "relay.Span": _rename("Span"),
+        "relay.GlobalVar": _rename("GlobalVar"),
+        "relay.Pass": _rename("transform.Pass"),
+        "relay.PassInfo": _rename("transform.PassInfo"),
+        "relay.PassContext": _rename("transform.PassContext"),
+        "relay.ModulePass": _rename("transform.ModulePass"),
+        "relay.Sequential": _rename("transform.Sequential"),
+        "relay.attrs.ResizeAttrs": _update_resize_attrs,
+        # TIR
+        "Variable": _update_tir_var("tir.Var"),
+        "SizeVar": _update_tir_var("tir.SizeVar"),
+    }
+    return create_updater(node_map, "0.6", "0.7")
+
+def check_tensorrt_compatibility(file_path):
```

Comment:
I think this is a temporary solution it seems not quite safe to just update the strings as they may change again and there might be more nodes. For the long term, we should probably retire the old version or ask for update to the newer version.

Reply:
Hi Zhi, I agree. This is a temporarily solution to allow some degree of backwards compatibility until we can include DLR runtime with the compiled artifacts.